xarray/tests/test_dataset.py

```diff
@@ -5654,6 +5654,37 @@ def test_coarsen_coords_cftime():
     np.testing.assert_array_equal(actual.time, expected_times)
 
 
+def test_coarsen_keep_attrs():
+    _attrs = {"units": "test", "long_name": "testing"}
+
+    var1 = np.linspace(10, 15, 100)
+    var2 = np.linspace(5, 10, 100)
+    coords = np.linspace(1, 10, 100)
+
+    ds = Dataset(
+        data_vars={"var1": ("coord", var1), "var2": ("coord", var2)},
+        coords={"coord": coords},
+        attrs={'units':'test', 'long_name':'testing'}
+    )
+
+    # Test dropped attrs
+    dat = ds.coarsen(coord=5).mean()
+    assert dat.attrs == {}
+
+    # Test kept attrs using dataset keyword
+    dat = ds.coarsen(coord=5, keep_attrs=True).mean()
+    assert dat.attrs == _attrs
+
+    # # Test kept attrs using wrapper function keyword
+    # dat = ds.coarsen(coord=5).mean(keep_attrs=True)
```

Comment:
Yeah, I don't think this should work; `keep_attrs` needs to be passed specifically to `coarsen`

Reply:
I am still building intuition for this package so thank you for clarifying. I will remove the comment.